File: simulators/de.unistuttgart.iste.sqa.mpw.framework.java/core/src/main/java/de/unistuttgart/iste/sqa/mpw/framework/FrameworkAssert.java
package de.unistuttgart.iste.sqa.mpw.framework;

public class FrameworkAssert {

	private FrameworkAssert() {}

	public static void assertCondition(final boolean condition, final String message) {
		if (!condition) {
			throw new RuntimeException(message);
		}
	}

	public static void assertAttributeCondition(final String condition, final boolean valid) {
		if (!valid) {
			throw new RuntimeException("Attribute condition is not valid: " + condition);
		}
	}
	
	public static void assertInstanceNotNull(final String instanceName, final Object instance) {
		if (instance == null) {
			throw new RuntimeException("Instance must not be null: " + instanceName);
		}
	}
	
	public static void assertObjectFound(final String referenceName, final String objectTypeName, final Object instance) {
		if (instance == null) {
			throw new RuntimeException("Object of type '" + objectTypeName + "' for reference '" + referenceName + "' not found");	
		}
	}

}

File: core/src/framework/FrameworkAssert.cpp
#include "FrameworkAssert.h"

#include <string>
#include "AssertionFailedException.h"

namespace framework {

void assertCondition(bool condition, const std::string& message) {
    if (!condition) {
        throw AssertionFailedException(message);
    }
}

void assertAttributeCondition(const std::string& condition, bool valid) {
    assertCondition(valid, "Attribute condition is not valid: " + condition);
}

void assertInstanceNotNull(const std::string& instanceName, EntityReference instance) {
    assertCondition(instance != nullptr, "Instance must not be null: " + instanceName);
}

void assertObjectFound(const std::string& referenceName, const std::string& objectTypeName,
                       EntityReference instance) {
    assertCondition(instance != nullptr,
                    "Object of type '" + objectTypeName + "' for reference '" + referenceName + "' not found");
}

void checkState(bool check) {
    if (!check) {
        throw AssertionFailedException();
    }
}

void checkState(bool check, const std::string& message) {
    if (!check) {
        throw AssertionFailedException(message);
    }
}

}
